snapcraft/parts/parts.py

```diff
@@ -102,6 +104,9 @@ def __init__(
                 project_name=project_name,
                 project_vars_part_name=adopt_info,
                 project_vars=project_vars,
+                # custom arguments
+                project_base=project_base,
```

Comment:
This is needed because Craft-parts expects `base` to be the effective base (i.e. build-base if base is bare).

Reply:
I tried other names and found them to be slightly confusing. The value originates from `project.base` so keeping the project prefix name makes the variable instantly recognizable.